gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard/checkuplink

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+xor2() {
+        echo -n "${1:0:1}"
+        echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
+}
+
+interface_linklocal() {
+        # We generate a predictable v6 address
+        local macaddr="$(echo $(uci get wireguard.wireguard.privatekey | wg pubkey) |md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
+        local oldIFS="$IFS"; IFS=':'; set -- $macaddr; IFS="$oldIFS"
+        echo "fe80::$(xor2 "$1")$2:$3ff:fe$4:$5$6"
+}
+
+#Wurde bereits ein privatekey generiert?
+temp=$(uci get wireguard.wireguard.privatekey);
+#Falls nicht, wird das hier nachgeholt
+if [ "$?" -ne "0" ]; then
+        uci set wireguard.wireguard.privatekey=$(wg genkey);
+        uci commit wireguard
+fi
```

Comment:
We should do this on firstboot / when rebooting. This way, we can display the public key on the reboot page and support whitelisted communities.

For fastd this happens on reboot and there was a specific rationale for this, not sure if this applies here.

Reply:
Yep, that makes sense.